src/identity/mod.rs

```diff
@@ -0,0 +1,111 @@
+mod platform;
+
+use crate::config::inputs;
+use failure::{Fallible, ResultExt};
+use serde::Serialize;
+use std::collections::HashMap;
+
+/// Kernel arguments location
+static KERNEL_ARGS_FILE: &str = "/proc/cmdline";
+
+/// Agent identity.
+#[derive(Debug, Serialize)]
+pub(crate) struct Identity {
+    /// Collecting level
+    pub(crate) level: String,
+    /// OS platform.
+    pub(crate) platform: String,
+}
+
+impl Identity {
+    /// Create from configuration.
+    pub(crate) fn new(cfg: &inputs::CollectingInput) -> Fallible<Self> {
+        let collecting_level = &cfg.level;
+        let id = match collecting_level.as_str() {
+            "minimal" => Self::try_default("minimal").context("failed to build minimal (default) identity")?,
+            "full" => Self::try_default("full").context("failed to build full identity")?,
+            &_ => Self::try_default("minimal").context("failed to build minimal (default) identity")?,
+        };
+
+        Ok(id)
+    }
+
+    /// Try to fetch default data
+    pub fn try_default(level: &str) -> Fallible<Self> {
+        let platform = platform::read_id(KERNEL_ARGS_FILE)?;
+
+        let id = match level {
+                    "minimal" => Self {
+                                    level: String::from("minimal"),
+                                    platform,
+                                },
+                    "full" => Self {
+                                    level: String::from("full"),
+                                    platform
+                                },
+                    &_ => Self {
+                                    level: String::from("minimal"),
+                                    platform,
+                                },
+                };
+
+        Ok(id)
+    }
+
+    /// Getter for collected metrics, returned as a HashMap
+    pub fn get_metrics(&self) -> HashMap<String, String> {
+        let mut vars = HashMap::new();
```

Comment:
In case you don't know it, check https://docs.rs/maplit/1.0.2/maplit/macro.hashmap.html. It could be handy here I guess.

Reply:
Ah, that's better!